packages/ui/src/components/accordion.tsx

```diff
@@ -0,0 +1,95 @@
+  ({ onValueChange, ...props }, ref) => <AccordionPrimitive.Root ref={ref} {...props} onValueChange={onValueChange} />
+)
+Accordion.displayName = 'Accordion'
+
+type AccordionItemProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Item> & {
+  isLast?: boolean
+}
+
+const AccordionItem = React.forwardRef<React.ElementRef<typeof AccordionPrimitive.Item>, AccordionItemProps>(
+  ({ className, isLast = false, ...props }, ref) => (
+    <AccordionPrimitive.Item ref={ref} className={cn('border-b', { 'border-b-0': isLast }, className)} {...props} />
+  )
+)
+AccordionItem.displayName = 'AccordionItem'
+
+const ChevronIcon = ({ chevronClass }: { chevronClass?: string }) => {
+  return (
+    <Icon
+      name="chevron-down"
+      size={10}
+      className={cn('chevron-down text-icons-2 h-2.5 w-2.5 min-w-2.5 shrink-0', chevronClass)}
+    />
+  )
+}
+
+type AccordionTriggerProps = React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Trigger> & {
+  hideChevron?: boolean
+  leftChevron?: boolean
+  rotateChevron?: boolean
+  chevronClass?: string
+}
+
+const AccordionTrigger = React.forwardRef<React.ElementRef<typeof AccordionPrimitive.Trigger>, AccordionTriggerProps>(
+  (
+    {
+      className,
+      hideChevron = false,
+      leftChevron = false,
+      rotateChevron = 'false',
+      chevronClass = '',
+      children,
+      ...props
+    },
+    ref
+  ) => (
+    <AccordionPrimitive.Header className="flex">
+      <AccordionPrimitive.Trigger
+        ref={ref}
+        className={cn(
+          'group flex flex-1 items-center justify-between py-4 text-sm font-medium transition-all',
+          '[&>svg]:duration-100 [&>svg]:ease-in-out [&>svg]:data-[state=open]:rotate-180',
+          {
+            'cursor-default': hideChevron,
+            'gap-1.5': leftChevron,
+            '[&>svg]:-rotate-90 [&>svg]:data-[state=open]:-rotate-0': rotateChevron
+          },
+          className
+        )}
+        {...props}
+      >
+        {leftChevron && !hideChevron && <ChevronIcon chevronClass={chevronClass} />}
+        {children}
+        {!leftChevron && !hideChevron && <ChevronIcon chevronClass={chevronClass} />}
+      </AccordionPrimitive.Trigger>
+    </AccordionPrimitive.Header>
+  )
+)
+AccordionTrigger.displayName = AccordionPrimitive.Trigger.displayName
+
+const AccordionContent = React.forwardRef<
+  React.ElementRef<typeof AccordionPrimitive.Content>,
+  React.ComponentPropsWithoutRef<typeof AccordionPrimitive.Content>
+>(({ className, children, ...props }, ref) => (
+  <AccordionPrimitive.Content
+    ref={ref}
+    className="overflow-hidden text-sm data-[state=closed]:animate-accordion-up data-[state=open]:animate-accordion-down"
+    {...props}
+  >
+    <div className={cn('pb-4 pt-0', className)}>{children}</div>
+  </AccordionPrimitive.Content>
+))
+AccordionContent.displayName = AccordionPrimitive.Content.displayName
+
+export { Accordion, AccordionItem, AccordionTrigger, AccordionContent }
```

Comment:
Might be handy to export the relevant types too

Reply:
This file contains only types related to the typing of the component's props. There are no other types that would be needed outside the file, so there is no type export here.

Reply:
Types for components are regularly used outside of the component themselves. For example, my favourite way of testing involves this...

```typescript
const renderComponent = (props: Partial<SomeComponentProps> = {}): RenderResult =>
  render(<SomeComponent prop1="default" prop2="default2" {...props} />)

test('it should show prop1', async () => {
  const prop1 = 'hello world'
  renderComponent({ prop1 })

  expect(await screen.findByText(prop1)).toBeInTheDocument()
})
```

You can extract a component's props using TS, but it's pain and most devs will have to Google to figure it out. Having the type exported with the component and with the naming convention of `${COMPONENT_NAME}Props` means they can get at it with no fuss.

This is a very common and encouraged approach in Harness and we need to continue it with this stuff.